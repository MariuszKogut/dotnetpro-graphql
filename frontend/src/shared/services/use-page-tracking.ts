import { useApplicationInsightsContext } from './analytics'

export const usePageTracking = () => {
  const { instance } = useApplicationInsightsContext()
  if (instance) {
    instance.trackPageView()
  }
}
