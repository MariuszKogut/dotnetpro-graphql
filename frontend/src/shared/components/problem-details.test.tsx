import React from 'react'
import { render } from '@testing-library/react'
import ProblemDetails from './problem-details'
import { IValidationProblemDetails } from '../services/customer-client'

describe('ProblemDetails', () => {
  const problemDetails: IValidationProblemDetails = {
    type: 'https://tools.ietf.org/html/rfc7231#section-6.5.1',
    title: 'One or more validation errors occurred.',
    status: 400,
    errors: {
      Name: [
        "'Name' darf nicht leer sein.",
        "Die LÃ¤nge von 'Name' muss zwischen 3 und 255 Zeichen liegen. Es wurden 0 Zeichen eingetragen.",
      ],
    },
  }

  it('renders empty list if no ProblemDetails given', () => {
    const { container } = render(<ProblemDetails fieldName="Fieldname" />)
    expect(container).toMatchSnapshot()
  })

  it('renders empty list if fieldName does not match', () => {
    const { container } = render(
      <ProblemDetails fieldName="Fieldname" problemDetails={problemDetails} />,
    )
    expect(container).toMatchSnapshot()
  })

  it('renders one item if fieldName matches', () => {
    const { container } = render(
      <ProblemDetails fieldName="Name" problemDetails={problemDetails} />,
    )
    expect(container).toMatchSnapshot()
  })
})
