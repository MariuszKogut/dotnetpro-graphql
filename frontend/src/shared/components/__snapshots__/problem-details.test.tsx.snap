// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblemDetails renders empty list if fieldName does not match 1`] = `<div />`;

exports[`ProblemDetails renders empty list if no ProblemDetails given 1`] = `<div />`;

exports[`ProblemDetails renders one item if fieldName matches 1`] = `
<div>
  <div
    class="invalid-feedback"
  >
    <span>
      'Name' darf nicht leer sein.
    </span>
    <span>
      Die LÃ¤nge von 'Name' muss zwischen 3 und 255 Zeichen liegen. Es wurden 0 Zeichen eingetragen.
    </span>
  </div>
</div>
`;
